Server.py

```diff
@@ -1,136 +1,146 @@
-server_socket.bind(('localhost', 12345))
-print("Connected to Server")
-server_socket.listen(5)
-
-try:
-    while True:
-        client_socket, client_address = server_socket.accept()
-        print(f"Connection from {client_address} has been established.")
-        client_handler = threading.Thread(target=handle_client, args=(client_socket,))
-        client_handler.start()
-finally:
-    server_socket.close()
+import socket
+import threading
+import Data
+
+DATABASE_FILE = "Pink_Floyd_DB.txt"
+
+albums = Data.parse_file(DATABASE_FILE)
+
+
+def getAlbums():
+    album_list = "\n".join(album['name'] for album in albums)
+    return "the albums list:\n" + album_list
+
+
+def getAlbumsSongs(albumName):
+    msg = "The songs in the album: \n"
+    for album in albums:
+        if album['name'].lower() == albumName.lower():
+            for song in album['songs']:
+                msg += f'{song['name']}\n'
+    return msg
+
+
+def getSongLength(songName):
+    msg = ""
+    found = False
+    for album in albums:
+        for song in album['songs']:
+            if songName.lower() == song['name'].lower():
+                msg = f'the Duration of {songName} is: {song['duration']}'
+                found = True
+    if not found:
+        return "Song not found"
+    return msg
+
+
+def getSongLyrics(songName):
+    msg = ""
+    found = False
+    for album in albums:
+        for song in album['songs']:
+            if songName.lower() == song['name'].lower():
+                found = True
+                msg = f'{song['lyrics']}'
+    if not found:
+        return "Song not found"
+    return msg
+
+
+def get_song_albums(songName):
+    msg = ""
+    found = False
+    for album in albums:
+        for song in album['songs']:
+            if songName.lower() == song['name'].lower():
+                found = True
+                msg = f'{album['name']}'
+    if not found:
+        return "Song not found"
+    return msg
+
+
+def SearchSongName(word):
+    msg = ""
+    found = False
+    for album in albums:
+        for song in album['songs']:
+            if word.lower() in song['name'].lower():
+                msg += f'{song['name']}\n'
+                found = True
+    if not found:
+        msg = "Song not found."
+    return msg
+
+
+def SearchSongLyrics(word):
+    msg = ""
+    found = False
+    for album in albums:
+        for song in album['songs']:
+            if word.lower() in song['lyrics'].lower():
+                msg += f'{song['name']}\n'
+                found = True
+    if not found:
+        msg = "Song not found."
+    return msg
+
+
+command_map = {
+    '1': getAlbums,
+    '2': (getAlbumsSongs,"Enter the album Name:"),
+    '3': (getSongLength,"Enter a song Name:"),
+    '4': (getSongLyrics,"Enter a song Name:"),
+    '5': (get_song_albums,"Enter a song name:"),
+    '6': (SearchSongName,"Enter a Word:"),
+    '7': (SearchSongLyrics,"Enter a Word:"),
+}
+
+
+def command_helper(socket, text: str, case: str):
```

Comment:
if use type hinting, use it for the entire code base (type hinting is a very good practice for clean code)